govcd/vm_test.go

```diff
@@ -450,10 +450,32 @@ func (vcd *TestVCD) Test_InsertOrEjectMedia(check *C) {
 	err = ejectMediaTask.WaitTaskCompletion()
 	check.Assert(err, IsNil)
 
-	//verify
 	err = vm.Refresh()
 	check.Assert(err, IsNil)
-	check.Assert(isMediaInjected(vm.VM.VirtualHardwareSection.Item), Equals, false)
+
+	// Expecting outcome to be 'false', but VCD sometimes lags to report that media is ejected in VM
+	// state (even after ejection task is finished).
+	// In such cases, do additional VM refresh and try to see if the structure has no media anymore
+	firstIsMediaNotEjected := isMediaInjected(vm.VM.VirtualHardwareSection.Item)
+	retryCount := 5
+	if firstIsMediaNotEjected {
+		// Run a few more attempts every second to see if the VM finally shows no media being
+		// present in VirtualHardwareItem structure
+		for a := 0; a < retryCount; a++ {
+			fmt.Printf("attempt %d\n", a)
+			err = vm.Refresh()
+			if err != nil {
+				fmt.Printf("error refreshing VM: %s\n", err)
+			}
+			loopisMediaNotEjected := isMediaInjected(vm.VM.VirtualHardwareSection.Item)
```

Comment:
`lopisMediaNotEjected` is a confusing name, because it contains `Not` and we compare it with `false` in the next line. 
Perhaps `loopIsMediaInjected` (and `if !loopIsMediaInjected` in the next line) ?

Reply:
Good point. Made an attempt to change names